src/dune_lang/package_dependency.ml

```diff
@@ -40,12 +40,45 @@ let encode { name; constraint_ } =
   | Some c -> pair Package_name.encode Package_constraint.encode (name, c)
 ;;
 
+(* Check for common typos in package dependency constraints *)
+let check_for_typo ~loc { name; constraint_ } =
+  let open Package_constraint in
+  match constraint_ with
+  | Some (Uop (Relop.Eq, Value.String_literal "version")) ->
+    let message = 
+      User_message.make ~loc
+        [ Pp.textf
+            "Possible typo in constraint for dependency %S: '(= version)' might be a mistake."
+            (Package_name.to_string name)
+        ]
+        ~hints:[ Pp.text "Did you mean to use the `:version` variable instead? Example: (depends (bar :version))" ]
```

Comment:
Instead of `(depends (bar :version))`, it would be better to print the actual package name from the user's dune-project. Also `(depends (bar :version))` isn't the right syntax; it should be `(depends (bar (= :version)))`.